samples/msal-node-samples/auth-code-distributed-cache/src/AuthProvider.ts

```diff
@@ -0,0 +1,198 @@
+            },
+            system: {
+                loggerOptions: {
+                    loggerCallback(loglevel, message, containsPii) {
+                        console.log(message);
+                    },
+                    logLevel: LogLevel.Verbose,
+                    piiLoggingEnabled: false
+                },
+                // proxyUrl: "http://localhost:8888" // uncomment to capture traffic with Fiddler
+            }
+        } as Configuration;
+
+        const msalConfigWithMetadata = await AuthProvider.getMetadata(msalConfig, cacheClient);
+
+        return new AuthProvider(msalConfigWithMetadata, cacheClient);
+    }
+
+    private getMsalInstance(sessionId: string): ConfidentialClientApplication {
+        return new ConfidentialClientApplication({
+            ...this.msalConfig,
+            cache: {
+                cachePlugin: new DistributedCachePlugin(
+                    this.cacheClientWrapper,
+                    new PartitionManager(this.cacheClientWrapper, sessionId) // partitionKey homeAccountId
+                )
+            }
+        });
+    }
+
+    async prepareTokenRequest(authCodeUrlParameters: AuthorizationUrlRequest, sessionId: string, requestRoute?: string): Promise<{ authCodeUrl: string, state: string }> {
+
+        /**
+         * MSAL Node allows you to pass your custom state as state parameter in the Request object.
+         * The state parameter can also be used to encode information of the app's state before redirect.
+         * You can pass the user's state in the app, such as the page or view they were on, as input to this parameter.
+         */
+        const state = this.cryptoProvider.base64Encode(
+            JSON.stringify({
+                csrfToken: this.cryptoProvider.createNewGuid(), // create a GUID for csrf
+                redirectTo: requestRoute || '/',
+            })
+        );
+
+        const msalInstance = this.getMsalInstance(sessionId);
+
+        try {
+            const authCodeUrl = await msalInstance.getAuthCodeUrl({
+                ...authCodeUrlParameters,
+                state,
+            });
+
+            return { authCodeUrl, state };
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async getTokenInteractive(tokenRequest: AuthorizationCodeRequest, authCodePayLoad: AuthorizationCodePayload, sessionId: string): Promise<AuthenticationResult | null> {
+        const msalInstance = this.getMsalInstance(sessionId);
+        let tokenResponse = null;
+
+        try {
+            performance.mark("acquireTokenByCode-start");
+            await msalInstance.getTokenCache().getAllAccounts(); // required for triggering beforeCacheACcess
+            tokenResponse = await msalInstance.acquireTokenByCode(tokenRequest, authCodePayLoad);
+            performance.mark("acquireTokenByCode-end");
+            performance.measure("acquireTokenByCode", "acquireTokenByCode-start", "acquireTokenByCode-end");
+        } catch (error) {
```

Comment:
Avoid catch / throw

Reply:
my bad, must have been thinking of something else -removed